Gateway/Response/PaymentDetailsHandler.php

```diff
@@ -3,23 +3,58 @@
 
 use Magento\Payment\Gateway\Helper\SubjectReader;
 use Magento\Payment\Gateway\Response\HandlerInterface;
+use Magento\Framework\App\Filesystem\DirectoryList;
 
 class PaymentDetailsHandler implements HandlerInterface
 {
     /**
-     * @param string $url URL to Tesco Barcode generated in Omise Backend
-     * @return string Barcode in SVG format
+     * $var \Omise\Payment\Helper\OmiseHelper
      */
-    private function downloadPaymentFile($url) 
+    protected $_helper;
+
+    /**
+     * @var \Magento\Framework\Filesystem
+     */
+    protected $_filesystem;
+
+    /**
+     * @var string
+     */
+    protected $imageData;
+
+    /**
+     * @var string
+     */
+    protected $imageType;
+
+     /**
+     * @param \Omise\Payment\Helper\OmiseHelper $helper
+     * @param \Magento\Framework\Filesystem $filesystem $filesystem
+     */
+    public function __construct(
+        \Omise\Payment\Helper\OmiseHelper $helper,
+        \Magento\Framework\Filesystem $filesystem
+    ) {
+        $this->_helper = $helper;
+        $this->_filesystem = $filesystem;
+    }
+
+    /**
+     * @desc Gets image data and data type from given $url
+     * @param string $url URL to image generated in Omise Backend
+     * @return void
+     */
+    private function setPaymentFileData($url) 
     {
         $ch = curl_init();
         curl_setopt($ch, CURLOPT_URL, $url);
         curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
         curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
-
-        return curl_exec($ch);
+        $this->imageData = curl_exec($ch);
+        $this->imageType = curl_getinfo($ch, CURLINFO_CONTENT_TYPE);
+        curl_close($ch);
```

Comment:
✋  Is it possible to keep only the url of the QR code? (just wondering what's the benefit of make a Curl request here instead of save url and display it later on – just wonder in case Omise-WooCommerce plugin should apply this approach too)

Reply:
The Url which we get from the API is Url to action which returns the QR file i.e. not Url to the image directly. This is the reason stored file data and file type stored separately.